test/json_schemer_test.rb

```diff
@@ -129,6 +129,17 @@ def test_it_inserts_defaults
     }
   end
 
+  def test_it_does_not_fail_using_insert_defaults_when_the_schema_is_completely_empty
+    schema = {}
+    data = {
+      'a' => 1
+    }
+    assert JSONSchemer.schema(schema, insert_property_defaults: true).valid?(data)
```

Comment:
It's likely that `insert_property_defaults` isn't necessary to make this test fail. Can you try removing the option (and changing the test method name)?

Reply:
Thank you, I've made the requested changes and rebased off latest master